components/client/src/test/unit/java/com/hotels/styx/client/OriginsInventoryTest.java

```diff
@@ -85,207 +86,397 @@ public void stop() {
         logger.stop();
     }
 
+    /*
+     * Setting the origins
+     */
     @Test
-    public void addingANewOriginWillAddToActiveSetAndInitiateTheMonitoring() {
-        inventory.addOrigins(ORIGIN);
+    public void setNewOrigins() {
+        inventory.setOrigins(ORIGIN_1, ORIGIN_2);
+
+        assertThat(inventory, hasActiveOrigins(2));
+        verify(monitor).monitor(singleton(ORIGIN_1));
+        verify(monitor).monitor(singleton(ORIGIN_2));
+
+        assertThat(gaugeValue("origins.generic-app.app-01.status"), isValue(1));
+        assertThat(gaugeValue("origins.generic-app.app-02.status"), isValue(1));
+
+        verify(eventBus).post(any(OriginsInventorySnapshot.class));
+    }
+
+    @Test
+    public void updatesOriginPortNumber() throws Exception {
+        Origin originV1 = newOriginBuilder("acme.com", 80).applicationId(GENERIC_APP).id("acme-01").build();
+        Origin originV2 = newOriginBuilder("acme.com", 443).applicationId(GENERIC_APP).id("acme-01").build();
+
+        inventory.setOrigins(originV1);
 
         assertThat(inventory, hasActiveOrigins(1));
-        verify(monitor).monitor(singleton(ORIGIN));
+        verify(monitor).monitor(singleton(originV1));
+        assertThat(gaugeValue("origins.generic-app.acme-01.status"), isValue(1));
+        verify(eventBus).post(any(OriginsInventorySnapshot.class));
 
-        inventory.registerStatusGauges();
-        assertThat(gaugeValue("origins.generic-app.one.status"), isValue(1));
+        inventory.setOrigins(originV2);
+
+        assertThat(inventory, hasActiveOrigins(1));
+        verify(monitor).stopMonitoring(singleton(originV1));
+        verify(monitor).monitor(singleton(originV2));
+        assertThat(gaugeValue("origins.generic-app.acme-01.status"), isValue(1));
+        verify(eventBus, times(2)).post(any(OriginsInventorySnapshot.class));
     }
 
+    @Test
+    public void updatesOriginHostName() throws Exception {
```

Comment:
This name is also not as informative as it could be. Perhaps something like "replaces origin when host name has changed"?

Reply:
... again which is just a longer way saying "updates origin host name".